### ISSUE TITLE
0.32.5 - wrong color space / depth

### ISSUE THREAD

**Issue by RadekHavelka** [lovell/sharp#3773]
Hello, our code stoped working after upgrading from 0.32.1 to 0.32.5. I've found issue similar in previous versions

https://github.com/lovell/sharp/issues/3605

our code, relevant part

```
 const readStream = this.storage
      .bucket(bucketName)
      .file(sourceCloudFilename)
      .createReadStream();

    const writeStream = this.storage
      .bucket(bucketName)
      .file(targetCloudFilename)
      .createWriteStream();

const transformStreamSharp = sharp({
      sequentialRead: true,
      failOnError: false,
    })
      .withMetadata()
      .toColorspace('rgb16')
      .png({
        compressionLevel: 5,
      })
      .trim()
      .toBuffer((_err, _buffer, outputInfo) =>
        this.uploadTrimmingInfo(trimInfoCloudFilename, bucketName, outputInfo),
      );
      
      await streamPromises.pipeline(
      readStream,
      transformStreamSharp,
      writeStream,
    );
```

Expected result: image with 16bit color depth
Actual result: image with 8bit color depth

(input data are ofc the same, image with 16bpc)

I think the function ".toColorspace('rgb16')" has some issues in recent version.

**Comment by lovell** [comment]
Hi, thanks for reporting, this is a regression in v0.32.5 due to the change for #3761. This is fixed by commit https://github.com/lovell/sharp/commit/95635683acf698728f589e9e6e8544b1aa13622e that also adds a test to prevent further regression.

**Comment by 7freaks-otte** [comment]
@lovell are you sure this is related to https://github.com/lovell/sharp/issues/3761 and v0.32.5?

I am still on v0.32.4 and the following code correctly generates a 16bit sRGB PNG

```js
await sharp('test_p3-16bit.png')
  .pipelineColourspace('rgb16')
  .toColourspace('rgb16')
  //.withMetadata({ icc: 'p3' })
  .resize({ width: 800, height: 800 })
  .png({ progressive: false, force: true })
  .toFile('test_out.png');
```

But as soon as withMetadata is uncommented, the output file will be an **8bit** P3 PNG

**Comment by 7freaks-otte** [comment]
@lovell Thanks for the clarification, looking forward to seeing fixed this in the future. Any plans when this could arrive?

**Comment by lovell** [comment]
I've applied the equivalent of the upstream libvips fix to sharp ahead of libvips 8.15.0 being made available as doing so also simplifies the fix for the original issue - see follow-up commit https://github.com/lovell/sharp/commit/9c217ab580123ee14ad65d5043d74d8ea7c245e5

**Comment by lovell** [comment]
v0.32.6 now available, thanks for reporting.
